contentcuration/contentcuration/serializers.py

```diff
@@ -915,3 +917,23 @@ def get_channel_ids(self, channelset):
     class Meta:
         model = ChannelSet
         fields = ('id', 'name', 'description', 'public', 'editors', 'channels', 'secret_token')
+
+
+class TaskSerializer(serializers.ModelSerializer):
+    metadata = serializers.SerializerMethodField()
+
+    def get_metadata(self, task):
+        metadata = task.metadata
+        result = app.AsyncResult(task.id)
+        if not settings.CELERY_TASK_ALWAYS_EAGER:
```

Comment:
What's the reason we have two behaviours, depending on celery's eagerness?

Reply:
CELERY_TASK_ALWAYS_EAGER makes the task run synchronous, so the task has finished by the time this is called. Let me play with this a bit more and see if I can find a way to avoid having this code path.

Reply:
I can't remove the check, because this code path will assert if run when CELERY_TASK_ALWAYS_EAGER is set, but I was able to remove the hardcoded value for tests.